### PR TITLE
Starting on async tests

### PR DESCRIPTION
**Pull request by kevmoo**

### CODE REVIEW

File: pkgs/checks/lib/src/extensions/async.dart [hunk 1]
```diff
@@ -80,7 +80,7 @@ extension StreamChecks<T> on Check<StreamQueue<T>> {
   /// Expects that the `Stream` emits any number of events before emitting an
   /// event that satisfies [condition].
   ///
-  /// Returns a `Future` that completes after the stream has emitted an even
+  /// Returns a `Future` that completes after the stream has emitted an event
```
Comment (kevmoo) [inline pkgs/checks/lib/src/extensions/async.dart:83]: oops!

**Comment by kevmoo** [comment]
Inspiration to figure out testing failures in async code...could NOT figure it out!

File: pkgs/checks/test/extensions/async_test.dart [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+import 'package:async/async.dart';
+import 'package:checks/checks.dart';
+import 'package:test/scaffolding.dart';
+
+void main() {
+  group('FutureChecks', () {
+    test('completes', () async {
+      (await checkThat(_futureSuccess()).completes()).equals(42);
+
+      // TODO: validate the failure case
```
Comment (natebosch) [inline pkgs/checks/test/extensions/async_test.dart:14]: Hmm, we might need an async version of `softCheck` for this.

**Review by natebosch — APPROVED** [review]

File: pkgs/checks/test/extensions/async_test.dart (hunk 2 above)
Comment (natebosch) [reply pkgs/checks/test/extensions/async_test.dart:14]: Or perhaps we do something like

```dart
test('completes - fails for an error future', () async {
  Future<void> failingTest() async {
    await checkThat(_futureFail()).completes();
  }
  await checkThat(failingTest()).throw();
}
```

Comment (kevmoo) [reply pkgs/checks/test/extensions/async_test.dart:14]: All we get is a `TestFailure` exception – can't validate the guts, which I think we want!

Comment (natebosch) [reply pkgs/checks/test/extensions/async_test.dart:14]: Yeah, having the `Rejection` would be nicer. I thought that async `softCheck` would be challenging, but it turns out not to be.

https://github.com/dart-lang/test/pull/1789

**Pull request merged**